lms/djangoapps/experiments/flags.py

```diff
@@ -162,16 +180,28 @@ def get_bucket(self, course_key=None, track=True):
             if not self._is_enrollment_inside_date_bounds(values, user, course_key):
                 return self._cache_bucket(experiment_name, 0)
 
-        bucket = stable_bucketing_hash_group(experiment_name, self.num_buckets, user.username)
-
-        # Now check if the user is forced into a particular bucket, using our subordinate bucket flags
+        # Determine the user's bucket.
+        # First check if forced into a particular bucket, using our subordinate bucket flags.
+        # If not, calculate their default bucket using a consistent hash function.
         for i, bucket_flag in enumerate(self.bucket_flags):
-            if bucket_flag.is_enabled(course_key):
+            forced_into_bucket = (
+                bucket_flag.is_enabled(course_key) if course_key
+                else bucket_flag.is_enabled_without_course_context()
+            )
+            if forced_into_bucket:
```

Comment:
Not much value to the variable to carry itself forward one line, but I'm guessing you felt like it adds non-obvious commentary to the conditional?

Reply:
(diff obsolete with 2nd commit)